invenio_rdm_records/resources/decorators.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record Resource Decorators."""
+from functools import wraps
+
+from flask import Response
+from flask_resources import resource_requestctx
+
+
+def custom_response_handler(many=False):
+    """Decorator for using the response handler to create the HTTP response.
+
+    The response handler works in conjunction with ``with_content_negotiation()``
+    which is responsible for selecting the correct response handler based on
+    the content negotiation.It checks if the route function explicitly
+    returns a Flask Response. If so, the response is returned as is.
+
+    .. code-block:: python
+
+        @response_handler()
+        def read(self):
+            return obj, 200
+
+        @response_handler(many=True)
+        def search(self):
+            return [obj], 200
+    """
+
+    def decorator(f):
+        @wraps(f)
+        def inner(*args, **kwargs):
+            res = f(*args, **kwargs)
+
+            # Check if the result is already a Flask Response
+            if isinstance(res, Response):
+                return res
```

Comment:
I would say this should be added in the [`response_handler`](https://github.com/inveniosoftware/flask-resources/blob/master/flask_resources/responses.py#L18-L44)in Flask by default, I can crate a PR there unless I am missing something and this is not the best solution.

Reply:
Isn't it the case already [here](https://github.com/inveniosoftware/flask-resources/blob/2f74ca86df50501742b949aa397ab6d4408224b3/flask_resources/responses.py#L83)? Otherwise it also looks like Flask [checks here](https://github.com/pallets/flask/blob/9a12f34bbee68a14301bc234f50c4a9fea9003fd/src/flask/app.py#L1168).

Reply:
I feel like we've dug ourselves in a hole with `flask-resources`... I see a couple of options:

- Do something similar [like Flask does](https://github.com/pallets/flask/blob/9a12f34bbee68a14301bc234f50c4a9fea9003fd/src/flask/app.py#L1139), and start checking for the type of `res`. It feels like a lot of duplication tbh...
- Keep the convention and return both the response object and the code (even in this case of the redirect, i.e. `return redirect(...), 302`)